doc/src/manual/conversion-and-promotion.md

```diff
@@ -233,11 +233,11 @@ julia> promote(1 + 2im, 3//4)
 ```
 
 Floating-point values are promoted to the largest of the floating-point argument types. Integer
-values are promoted to the larger of either the native machine word size or the largest integer
-argument type. Mixtures of integers and floating-point values are promoted to a floating-point
-type big enough to hold all the values. Integers mixed with rationals are promoted to rationals.
-Rationals mixed with floats are promoted to floats. Complex values mixed with real values are
-promoted to the appropriate kind of complex value.
+values are promoted to the largest of the integer argument types. If the types are the same size
+but differ in signedness, the unsigned type is chosen. Mixtures of integers and floating-point
+values are promoted to a floating-point type big enough to hold all the values. Integers mixed
```

Comment:
The floating point comment doesn't seem particularly accurate: it is just promoted to the floating point type, whatever that is. For example:
```
julia> promote(Float16(1.0f0), typemax(Int))
(Float16(1.0), Inf16)

julia> promote(Float32(1), typemax(Int32))[2] |> Int 
2147483648

julia> typemax(Int32)
2147483647
```

Reply:
Correct @vtjnash. I can make the change to "Mixtures of integers and floating-point values are promoted to the floating-point type." if this is the wanted behavior and the current implementation is not the problem?

Reply:
My point is that some people might find this quite unexpected:

```julia
julia> max(Float16(1), typemax(Int64))
Inf16

julia> versioninfo()
Julia Version 1.8.5
Commit 17cfb8e65ea (2023-01-08 06:45 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.5.0)
  CPU: 10 × Apple M1 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, apple-m1)
  Threads: 8 on 8 virtual cores
Environment:
  JULIA_NUM_THREADS = 8
```